### PR TITLE
Data models

### PR DESCRIPTION
**Pull request by pantelisss**
## **Why?**
Data models for reward details
### **How?**
Declared all necessary models and endpoints for reward details screen
### **Testing**
Make sure the models are declared properly 
### **Additional context**
fixes fe-654

### CODE REVIEW

File: wxm-ios/DomainLayer/DomainLayer/Entities/Codables/Me/Network/NetworkDeviceRewardDetailsResponse.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+			case actualReward = "actual_reward"
+			case rewardScore = "reward_score"
+			case maxReward = "max_reward"
+		}
+	}
+
+	struct Annotation: Codable, Hashable {
+		public let score: Int?
+		public let summary: [RewardAnnotation]?
+	}
```
Comment (PavlosTze) [inline wxm-ios/DomainLayer/DomainLayer/Entities/Codables/Me/Network/NetworkDeviceRewardDetailsResponse.swift:80]: I think that's wrong, as we have an AnnotationSummary there which has a different format?

e.g.
```
"annotation_summary": [
    {
      "severity_level": "INFO",
      "group": "NO_WALLET",
      "title": "Minor Sensor Issues",
      "message": "Station is doing great! Minor issues detected during our data quality checks, that may be related to occasional sensor inaccuracies. That can happen once in a while.",
      "doc_url": "https://docs.weatherxm.com/project/rewards-troubleshooting#sensor-problems"
    }
  ]
```

**Review by PavlosTze — CHANGES_REQUESTED** [review]

File: wxm-ios/DomainLayer/DomainLayer/Entities/Codables/Me/Network/NetworkDeviceRewardDetailsResponse.swift (hunk 1 above)
Comment (pantelisss) [reply wxm-ios/DomainLayer/DomainLayer/Entities/Codables/Me/Network/NetworkDeviceRewardDetailsResponse.swift:80]: Not sure I understand. Do you mean that is better to rename the mode from `RewardAnnotation` to `AnnotationSummary`?

Comment (PavlosTze) [reply wxm-ios/DomainLayer/DomainLayer/Entities/Codables/Me/Network/NetworkDeviceRewardDetailsResponse.swift:80]: Hmm no you are right, my bad. Regarding the above though, the `score` isn't defined in the API response, right?

Comment (pantelisss) [reply wxm-ios/DomainLayer/DomainLayer/Entities/Codables/Me/Network/NetworkDeviceRewardDetailsResponse.swift:80]: 😩 score is removed now... 
There was in the initial mock response

**Review by PavlosTze — APPROVED** [review]

**Pull request merged**